### PR TITLE
Light ration refactor

### LINKED ISSUE #105: Duplicate Pen Nutrient Requirement Attributes
There are currently separate attributes for the pen class called 'avg_nutrient_rqmts' and 'avg_calf_nutrient_rqmts'. I believe we should only need avg_nutrient_rqmts and it will contain the calf nutrient requirements if calves are in the pen

### PR DESCRIPTION
**Pull request by JoeWaddell**
### Context

Issue(s) closed by this pull request: closes #105 

To keep this PR's size down, I branched off this branch to make a more complex part 2: https://github.com/RuminantFarmSystems/MASM/pull/1573

- [x] The [changelog](https://docs.google.com/spreadsheets/d/1U-3igxdstHTFb6k5dVcE5-x9IL0r6q18eUiwwytKZyA/edit#gid=0) has been updated.
- [x] Needs to address missing units in ration_supply_report_units (# TODO: Missing the following: "forage_NDF_percent", "metabolizable_protein") and nutrient_amount_units (TODO: Missing the following: "as_fed", "EE", "starch", "TDN", "DE", "calcium")
- [x] Needs SME double-checking of units, specifically that kg and g are correct, where appropriate.

### What

Mostly typing updates

New TypedDict to help with type hints and circular imports

Changed name of `type` in AvailableFeeds to `feed_type` for clarity.

Corrected units in `nutrient_amount_units` dict in `report_ration_interval_data` in `AnimalModuleReporter` class.

### Why

Old methods pre-typing needed to be updated.

Ideally speeding up the process, and breaking down a complex method to be more manageable.

### How

Reduces MyPy errors and improves clarity and documentation.

### Test plan

Updated affected unit tests.

### CODE REVIEW

**Review by allisterakun — APPROVED** [review]
Great PR, Joe! 
The changes are all straightforward, LGTM. Thank you for taking the time to update all the type hints, which significantly decreases the MyPr error count, and for renaming the properties to be more readable.

File: RUFAS/routines/animal/animal_module_reporter.py [hunk 1]
```diff
@@ -218,20 +218,20 @@ def report_ration_interval_data(cls, pen_list: List[Pen], feed: Feed, simulation
...
-                "phosphorus": MeasurementUnits.PERCENT_OF_DRY_MATTER.value,
-                "potassium": MeasurementUnits.PERCENT_OF_DRY_MATTER.value,
-                "N": MeasurementUnits.PERCENT_OF_DRY_MATTER.value,
                 "as_fed": MeasurementUnits.PERCENT.value,
-                "EE": MeasurementUnits.PERCENT_OF_DRY_MATTER.value,
-                "starch": MeasurementUnits.PERCENT_OF_DRY_MATTER.value,
-                "TDN": MeasurementUnits.PERCENT_OF_DRY_MATTER.value,
-                "DE": MeasurementUnits.PERCENT_OF_DRY_MATTER.value,
-                "calcium": MeasurementUnits.PERCENT_OF_DRY_MATTER.value,
+                "CP": MeasurementUnits.KILOGRAMS.value,
```
Comment (tomhuhh) [inline RUFAS/routines/animal/animal_module_reporter.py]: Should it be 'Kilograms per animal' to keep it consistent with 'dm'?

**Review by tomhuhh — APPROVED** [review]
Thanks Joe! I think it generally looks good to me. I left a couple comments. Thanks!

File: RUFAS/routines/animal/animal_module_reporter.py [hunk 2]
```diff
@@ -218,20 +218,20 @@ def report_ration_interval_data(cls, pen_list: List[Pen], feed: Feed, simulation
...
                 "dm": MeasurementUnits.KILOGRAMS_PER_ANIMAL.value,
-                "CP": MeasurementUnits.PERCENT_OF_DRY_MATTER.value,
-                "ADF": MeasurementUnits.PERCENT_OF_DRY_MATTER.value,
-                "NDF": MeasurementUnits.PERCENT_OF_DRY_MATTER.value,
-                "lignin": MeasurementUnits.PERCENT_OF_DRY_MATTER.value,
-                "ash": MeasurementUnits.PERCENT_OF_DRY_MATTER.value,
-                "phosphorus": MeasurementUnits.PERCENT_OF_DRY_MATTER.value,
-                "potassium": MeasurementUnits.PERCENT_OF_DRY_MATTER.value,
-                "N": MeasurementUnits.PERCENT_OF_DRY_MATTER.value,
                 "as_fed": MeasurementUnits.PERCENT.value,
```
Comment (tomhuhh) [inline RUFAS/routines/animal/animal_module_reporter.py]: kg

**Pull request merged**